spec/features/add_room_spec.rb

```diff
@@ -24,8 +20,12 @@
 end
 
 describe "Unauthenticated User" do
-  it "does not have the option to create a new room" do
+  it "must login before creating a new room" do
     page.visit root_path
-    expect(page).not_to have_content("Create a new room")
+    expect(page).to have_content("Login to create room")
+    fill_in("room[name]", with: "testing")
+    fill_in("room[choice]", with: "rspec")
+    click_on("Login to create room")
+    expect(page).to have_content("Create New Room")
```

Comment:
Did you get the form persistence to work here? If not it's completely fine. Not necessary for the app.